src/scores/utils.py

```diff
@@ -112,6 +124,97 @@ def gather_dimensions(  # pylint: disable=too-many-branches
     return reduce_dims
 
 
+def gather_dimensions2(
+    fcst: XarrayLike,
```

Comment:
Add a test that checks that this works with Dataset inputs. I think it should work find with Datasets.

Reply:
So it doesn't work for datasets. 
For example, if I create a dataset `ds` then `ds.dims` returns `Frozen({'station': 2})`.
Whereas for a data array `da`, `da.dims` returns `('station',)`.

See this documentation for the difference in behaviour https://docs.xarray.dev/en/stable/generated/xarray.Dataset.dims.html#xarray.Dataset.dims

For now I have changed typehints to only accept xr.DataArray.
If `gather_dimensions2` is to be used universally in scores, it will have to extract the dimensions as tuples from xr.Dataset and pd.Dataframe objects too. I think that is beyond the scope of this pull request, but happy to discuss.

Reply:
FYI, I've cross-referenced this comment to the issue concerned https://github.com/nci/scores/issues/125